lib/internal/async_hooks.js

```diff
@@ -243,27 +246,104 @@ function restoreActiveHooks() {
   active_hooks.tmp_fields = null;
 }
 
+function trackPromise(promise, parent, silent) {
+  const asyncId = getOrSetAsyncId(promise);
```

Comment:
This const can be moved into the `if (!silent && initHooksExist()) {}` block

Reply:
Never mind, I forgot about the side-effect. Going to think about ways to simplify `trackPromise` and `promiseHook` functions.